helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java

```diff
@@ -613,7 +613,12 @@ void reset() {
 
       MsgHandlerFactoryRegistryItem item = _hdlrFtyRegistry.get(msgType);
       if (item.factory() != null) {
-        item.factory().reset();
+        try {
+          item.factory().reset();
+        } catch (Exception ex) {
```

Comment:
I don't quite understand why we swallow exception here. This exception means user's reset has an issue, and can we just let the error go?

Reply:
Just wondering, is it better to log the backtrace as well for better debuggability?

Reply:
This is only called when we clean up. All the paths lead to the termination of the executor. The only result of not swallowing the exception is that the later cleanup steps will not happen.

Backtrace is a good point. I think I missed it unintentionally. Thanks for pointing this out.